docs/contributor/devenv_setup.md

```diff
@@ -307,3 +307,108 @@ sudo lvextend -l +95%FREE /dev/mapper/rhel-root
+machine and run on the host, etc.).
+
+One way of sharing the directory is to use NFS mounts. Follow the instructions
+in the remainder of this section to implement the directory mount.
+
+#### Host Configuration
+
+Run the following commands to create and export the directory over an NFS mount.
+
+```
+EXPORT_DIR=/home/microshift-shared
+
+sudo dnf install -y nfs-utils
+
+sudo mkdir -p "${EXPORT_DIR}"
+sudo chown nobody:nobody "${EXPORT_DIR}"
+sudo chmod 755 "${EXPORT_DIR}"
+
+echo "${EXPORT_DIR} *(rw,async,no_root_squash,no_subtree_check)" | sudo tee -a /etc/exports
+sudo exportfs -arv
+```
+
+Run the following commands to configure the NFS service and firewall rules.
+
+```
+sudo systemctl restart nfs-server
+sudo systemctl enable nfs-server
+
+sudo firewall-cmd --permanent --zone=libvirt --add-service=nfs
+sudo firewall-cmd --permanent --zone=libvirt --add-service=mountd
+sudo firewall-cmd --permanent --zone=libvirt --add-service=rpc-bind
+sudo firewall-cmd --reload
+```
+
+Run the following commands to note down the IP address to be used when mounting
+the directory from the virtual machine.
+
+```
+BRIDGE_DEV=$(sudo virsh net-info default | grep '^Bridge:' | awk '{print $2}')
+BRIDGE_IP=$(ip -f inet addr show "${BRIDGE_DEV}" | grep inet)
+
+echo "${BRIDGE_IP}" | awk '{print $2}' | cut -d/ -f1
+```
+
+#### Virtual Machine Configuration
+
+```
+MOUNT_DIR=/home/microshift-shared
+BRIDGE_IP=192.168.100.1
+
+sudo dnf install -y nfs-utils
+
+sudo mkdir -p "${MOUNT_DIR}"
+sudo mount -t nfs4 -o context="system_u:object_r:container_file_t:s0" "${BRIDGE_IP}:${MOUNT_DIR}" "${MOUNT_DIR}"
+```
+
+If the mount works properly, make it permanent by running the following command.
+
+```
+echo "${BRIDGE_IP}:${MOUNT_DIR} ${MOUNT_DIR} nfs4 rw,hard,intr,noatime,context=\"system_u:object_r:container_file_t:s0\" 0 0" | sudo tee -a /etc/fstab
+```
+
+#### Synchronize User and Group Identifiers
+
+For read-write to work properly both on the hypervisor host and on the virtual
+machine, it is recommended to edit the `microshift` user and group
+identifiers on the virtual machine to have the value as on the hypervisor.
+
+Edit the `/etc/passwd` and `/etc/group` files on the virtual machine to replace
+the `microshift` user and group identifiers.
+
+Run the following commands to fix the ownership of the user home directory and
+the shared directory.
+
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo chown -R $(whoami). ~
+sudo chown -R $(whoami). "${MOUNT_DIR}"
+
+# Delete user-specific container cache
+sudo rm -rf ~/.local/share/containers/
+```
+
+Reboot the development virtual machine to finalize the configuration.
+
+#### Using Shared Directory
+
+One way of using the shared directory would be to keep the whole source tree in
+it for easy access from the host.
+
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo mkdir -p -m 0755 "${MOUNT_DIR}/microshift"
+sudo chown $(whoami). "${MOUNT_DIR}/microshift"
+
+ln -s "${MOUNT_DIR}/microshift" ~/microshift
```

Comment:
I think this misses git clone? Or at least move of the repo?
I already have source in ~/microshift, so the last command creates a symlink at `/home/microshift/microshift/microshift` pointing to `/home/microshift-shared/microshift` so it's invalid.

Also is it a bit weird that dev vm populates host's share? Why not share the microshift repo from the host directly mounted as ~/microshift in the VM?

Reply:
Having the whole home directory remotely mounted maybe a bit too intrusive. I have not tested it, but I can add a note that this is another sharing option.